File: programa4.m
try

n=input('Introduzca un número: ');
L=1:n;
total = 0;
if nnz(rem(n,L)==0)==2
    disp('Número primo');
    total = ('Numero primo');
else
    disp('Número compuesto');
    total = ('Numero Compuesto')
end

catch
 disp('algo salio mal');
 total = ('algo salio mal');
end_try_catch

pkg load database
conn = pq_connect (setdbopts ("dbname", "postgres", "host", "localhost", "port", "5432", "user", "postgres", "password", "3062887960313"))
N=pq_exec_params(conn, "insert into p4 values ($1,$2);",{'Programa 1',total});
N=pq_exec_params(conn, 'select * from p4;') %ver datos en la tabla

File: programa4.py
import psycopg2
try: 
    conexion = psycopg2.connect(
        host = "localhost",
        port = "5432",
        user = "postgres",
        password = "3062887960313",
        dbname = "postgres"
    )
    print("Conectado")
except psycopg2.Error as e:
    print("Ocurrió un error en la conexión")
    print("Verifique")


n=int(input("Ingrese un número: "))
cursor = conexion.cursor()
if n <= 0:
    print("El número debe ser mayor que cero")
else:
    cant_divisores = 0
    i = 1
    while (i <= n):
        if n % i == 0:
            cant_divisores+=1
        i+=1
    if cant_divisores==2:
        print("El número es primo")
        res = "Número primo"
        cursor.execute("insert into p4(numero, respuesta) values(%s, %s);",(n, res))
        conexion.commit()
    else:
        print("El número es compuesto")
        res = "Número compuesto"
        cursor.execute("insert into p4(numero, respuesta) values(%s, %s);",(n, res))
        conexion.commit()
        cursor.close()
        conexion.close()